BI Projekt - Topmotive

Zugriff auf die Daten:

In [2]:
import pandas as pd

In [5]:
df_Verrechnung = pd.read_csv('TM_WA_Verrechnungssätze.csv', on_bad_lines='skip', encoding='ISO-8859–1', delimiter=";")
df_Verrechnung

LocalID AWArt  \
0      0007BD7B-6934-4893-B971-CCA4857B1789    EL   
1      0007BD7B-6934-4893-B971-CCA4857B1789    KA   
2      0007BD7B-6934-4893-B971-CCA4857B1789    LA   
3      0007BD7B-6934-4893-B971-CCA4857B1789   NaN   
4      0007BD7B-6934-4893-B971-CCA4857B1789    WS   
...                                     ...   ...   
12059  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7    VL   
12060  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7    VM   
12061  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7    WS   
12062  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7   NaN   
12063  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7    ZB   

                            AWArt_Bez VKPreis VKPreis_Brutto EKPreis  \
0                    Elektrikarbeiten   12,40          14,76    6,80   
1                  Karosseriearbeiten   13,00          15,47    6,80   
2                     Lackierarbeiten   13,00          15,47    6,80   
3      Verrechnungssatz Autohaus Zorn   85,71         102,00    6,90   
4                   Werkstattarbeiten   10,30          12,26    6,30   
...                               ...     ...            ...     ...   
12059               Versicherung Lack  165,50         196,95  165,50   
12060           Versicherung Mechanik  135,50         161,25  135,50   
12061               Werkstattarbeiten  110,50         131,50    0,00   
12062             Werkstattarbeiten 2   59,95          71,34   59,95   
12063                 Zubehörarbeiten  135,95         161,78  135,95   

      AWMinuten               c_date  
0          6,00  01.07.2023 11:29:12  
1          6,00  01.07.2023 11:29:12  
2          6,00  01.07.2023 11:29:12  
3          6,00  01.07.2023 11:29:12  
4          6,00  01.07.2023 11:29:12  
...         ...                  ...  
12059     60,00  01.08.2023 05:59:20  
12060     60,00  01.08.2023 05:59:20  
12061     60,00  01.08.2023 05:59:20  
12062     60,00  01.08.2023 05:59:20  
12063     60,00  01.08.2023 05:59:20  

[12064 rows x 8 columns]

In [8]:
df_Belege = pd.read_csv('TM_WA_Belege.csv', on_bad_lines='skip', encoding='ISO-8859–1', delimiter=";")
df_Belege

/Users/justyna/opt/anaconda3/envs/master_bi/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (20) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


LocalID BELEG_ART  BELEGDATUM  KTypNr  \
0        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230719.0   25509   
1        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230719.0   25509   
2        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230719.0   25509   
3        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230601.0   29990   
4        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230601.0   29990   
...                                       ...       ...         ...     ...   
1373233  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373234  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373235  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373236  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373237  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   

        Fzg_Hersteller  ERSTZULASSUNG  KMSTAND POS_ART  \
0                 AUDI     20080801.0   208900       E   
1                 AUDI     20080801.0   208900       E   
2                 AUDI     20080801.0   208900       E   
3                   VW     20090415.0    81982       E   
4                   VW     20090415.0    81982       E   
...                ...            ...      ...     ...   
1373233            BMW            NaN       50       E   
1373234            BMW            NaN       50       E   
1373235            BMW            NaN       50       E   
1373236            BMW            NaN       50       E   
1373237            BMW            NaN       50       E   

                                               TEXT1  GENARTNR  ...  \
0                      MOTUL MULTI HF HYDRAULIKÖL 1L       NaN  ...   
1              TEXTAR BREMSBELAGSATZ, SCHEIBENBREMSE     402.0  ...   
2                                         E TEILE VW       NaN  ...   
3                                     BOSCH ÖLFILTER       7.0  ...   
4                                 MOTORÖL MOTUL 5W40    3224.0  ...   
...                                              ...       ...  ...   
1373233  EXT. BATTERIELADUNG ÜBER DEFA.MINIPLUG 230V       NaN  ...   
1373234                    LADEKABEL SCHUKO --> DEFA       NaN  ...   
1373235  FUNKHAUPTSCHALTER INTGT. IN DIE BEDIENEINH.       NaN  ...   
1373236              TONFOLGEANLAGE MODEL: „TFA 724“       NaN  ...   
1373237                                        FUGEN       NaN  ...   

        RechEmpfPLZ RechEmpfAnrede RechEmpfhatEMail RechEmpfhatMobil  \
0             53332          Herrn             NEIN             NEIN   
1             53332          Herrn             NEIN             NEIN   
2             53332          Herrn             NEIN             NEIN   
3             50997          Herrn             NEIN             NEIN   
4             50997          Herrn             NEIN             NEIN   
...             ...            ...              ...              ...   
1373233       01330          Firma               JA             NEIN   
1373234       01330          Firma               JA             NEIN   
1373235       01330          Firma               JA             NEIN   
1373236       01330          Firma               JA             NEIN   
1373237       01330          Firma               JA             NEIN   

        RechEmpfmitGeburtstag Belegerstelltam  Beleggedrucktam  \
0                        NEIN      20200512.0       20230719.0   
1                        NEIN      20200512.0       20230719.0   
2                        NEIN      20200512.0       20230719.0   
3                        NEIN      20210429.0       20230601.0   
4                        NEIN      20210429.0       20230601.0   
...                       ...             ...              ...   
1373233                  NEIN      20230731.0              NaN   
1373234                  NEIN      20230731.0              NaN   
1373235                  NEIN      20230731.0              NaN   
1373236       

In [9]:
df_PosArt = pd.read_csv('TM_WA_Belege.csv', on_bad_lines='skip', encoding='ISO-8859–1', delimiter=";")
df_PosArt

/Users/justyna/opt/anaconda3/envs/master_bi/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: DtypeWarning: Columns (20) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


LocalID BELEG_ART  BELEGDATUM  KTypNr  \
0        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230719.0   25509   
1        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230719.0   25509   
2        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230719.0   25509   
3        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230601.0   29990   
4        0007BD7B-6934-4893-B971-CCA4857B1789         R  20230601.0   29990   
...                                       ...       ...         ...     ...   
1373233  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373234  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373235  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373236  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   
1373237  FFD3E762-C6F5-4E9E-B5F2-5DE4B1B6D5D7         R  20230731.0   17814   

        Fzg_Hersteller  ERSTZULASSUNG  KMSTAND POS_ART  \
0                 AUDI     20080801.0   208900       E   
1                 AUDI     20080801.0   208900       E   
2                 AUDI     20080801.0   208900       E   
3                   VW     20090415.0    81982       E   
4                   VW     20090415.0    81982       E   
...                ...            ...      ...     ...   
1373233            BMW            NaN       50       E   
1373234            BMW            NaN       50       E   
1373235            BMW            NaN       50       E   
1373236            BMW            NaN       50       E   
1373237            BMW            NaN       50       E   

                                               TEXT1  GENARTNR  ...  \
0                      MOTUL MULTI HF HYDRAULIKÖL 1L       NaN  ...   
1              TEXTAR BREMSBELAGSATZ, SCHEIBENBREMSE     402.0  ...   
2                                         E TEILE VW       NaN  ...   
3                                     BOSCH ÖLFILTER       7.0  ...   
4                                 MOTORÖL MOTUL 5W40    3224.0  ...   
...                                              ...       ...  ...   
1373233  EXT. BATTERIELADUNG ÜBER DEFA.MINIPLUG 230V       NaN  ...   
1373234                    LADEKABEL SCHUKO --> DEFA       NaN  ...   
1373235  FUNKHAUPTSCHALTER INTGT. IN DIE BEDIENEINH.       NaN  ...   
1373236              TONFOLGEANLAGE MODEL: „TFA 724“       NaN  ...   
1373237                                        FUGEN       NaN  ...   

        RechEmpfPLZ RechEmpfAnrede RechEmpfhatEMail RechEmpfhatMobil  \
0             53332          Herrn             NEIN             NEIN   
1             53332          Herrn             NEIN             NEIN   
2             53332          Herrn             NEIN             NEIN   
3             50997          Herrn             NEIN             NEIN   
4             50997          Herrn             NEIN             NEIN   
...             ...            ...              ...              ...   
1373233       01330          Firma               JA             NEIN   
1373234       01330          Firma               JA             NEIN   
1373235       01330          Firma               JA             NEIN   
1373236       01330          Firma               JA             NEIN   
1373237       01330          Firma               JA             NEIN   

        RechEmpfmitGeburtstag Belegerstelltam  Beleggedrucktam  \
0                        NEIN      20200512.0       20230719.0   
1                        NEIN      20200512.0       20230719.0   
2                        NEIN      20200512.0       20230719.0   
3                        NEIN      20210429.0       20230601.0   
4                        NEIN      20210429.0       20230601.0   
...                       ...             ...              ...   
1373233                  NEIN      20230731.0              NaN   
1373234                  NEIN      20230731.0              NaN   
1373235                  NEIN      20230731.0              NaN   
1373236       

Statistik zu den Daten durchführen:

count: Anzahl der nicht fehlenden (nicht-NA/null) Einträge.
mean: Der Durchschnittswert der Einträge.
std: Die Standardabweichung, die die Streuung der Daten um den Durchschnitt herum misst.
min: Der kleinste Wert.
25%: Das 25. Perzentil, d.h. 25% der Daten liegen unter diesem Wert.
50%: Das Median oder 50. Perzentil.
75%: Das 75. Perzentil, d.h. 75% der Daten liegen unter diesem Wert.
max: Der größte Wert.

In [21]:
print("Verrechnung")
print(df_Verrechnung.describe())

Verrechnung
                                     LocalID AWArt          AWArt_Bez VKPreis  \
count                                  12064  7985              12064   12059   
unique                                  1007   122               1425    1155   
top     86E3ED88-C918-4BCD-9BA1-E8F38E8A4560    WS  Werkstattarbeiten   95,00   
freq                                     147  1748               1683     388   

       VKPreis_Brutto EKPreis AWMinuten               c_date  
count           12059   12059     12059                12064  
unique           1183     220         7                 1877  
top            113,05    0,00     60,00  01.08.2023 06:42:22  
freq              413    9908      7909                   53  


Bei der Tabelle Verrechnung konnte beobachetet werden:

1. Für "AWArt" gibt es nur 7985 Einträge, was bedeutet, dass es einige fehlende Daten in dieser Spalte gibt.
2. Einige Spalten für "LocalID" sind nicht eindeutig (da "unique" weniger ist als "count"). Das könnte ein Zeichen für redundante Daten oder Fehler sein.
3. Die Spalte "EKPreis" hat einen häufigsten Wert von "0,00", der 9908 Mal auftritt. Dies könnte darauf hinweisen, dass viele Artikel kostenlos sind oder dass fehlende/unaufgefüllte Werte als "0,00" markiert sind.

In [17]:
print("Belege")
print(df_Belege.describe())

Belege
         BELEGDATUM        KTypNr  ERSTZULASSUNG       KMSTAND       GENARTNR  \
count  1.347352e+06  1.373238e+06   1.314302e+06  1.373238e+06  535674.000000   
mean   2.022044e+07  4.307360e+04   2.009348e+07  1.296681e+05    2233.985114   
std    3.878877e+04  4.458935e+04   5.927797e+05  1.357906e+05    6110.950194   
min    1.995070e+07  0.000000e+00   1.010100e+04  0.000000e+00   -7528.000000   
25%    2.023061e+07  1.132500e+04   2.007100e+07  6.109300e+04     135.000000   
50%    2.023063e+07  2.269500e+04   2.012070e+07  1.141370e+05     654.000000   
75%    2.023071e+07  5.906800e+04   2.016063e+07  1.748540e+05    3059.750000   
max    2.023080e+07  1.533690e+05   9.091041e+07  9.977387e+06   85021.000000   

                PLZ  Fzg_anlage_datum  Belegerstelltam  Beleggedrucktam  \
count  1.305237e+06      1.309302e+06     1.307576e+06     1.220883e+06   
mean   4.276390e+04      2.019603e+07     2.022572e+07     2.023020e+07   
std    2.409396e+04      5.556088e+04 

Bei der Tabelle Belege konnte beobachetet werden:

1. GENARTNR: Diese Spalte hat negative Werte als Minimum und recht hohe positive Werte als Maximum. Es wäre wichtig,  zu überprüfen, ob negative Werte sinnvoll sind.
2. BELEGDATUM, ERSTZULASSUNG sind Datumsangaben, die als Ganzzahlen formatiert sind (z.B. 20220404 für den 4. April 2022)
3. KMSTAND: Das Maximum für "KMSTAND" ist extrem hoch (etwa 10 Millionen). Es könnte sinnvoll sein, solche extremen Werte genauer zu untersuchen, da sie Ausreißer oder Fehler sein könnten.
4. Programmvariant: Fast alle Werte in dieser Spalte scheinen "100" zu sein (aber es gibt einige wenige andere Werte das Maximum ist 104)
5. Fehlende Werte: Nicht alle Spalten haben den gleichen count, was darauf hinweist, dass es fehlende Werte gibt. Zum Beispiel hat "GENARTNR" nur 535674 Einträge


In [18]:
print("PosArt")
print(df_PosArt.describe())

PosArt
         BELEGDATUM        KTypNr  ERSTZULASSUNG       KMSTAND       GENARTNR  \
count  1.347352e+06  1.373238e+06   1.314302e+06  1.373238e+06  535674.000000   
mean   2.022044e+07  4.307360e+04   2.009348e+07  1.296681e+05    2233.985114   
std    3.878877e+04  4.458935e+04   5.927797e+05  1.357906e+05    6110.950194   
min    1.995070e+07  0.000000e+00   1.010100e+04  0.000000e+00   -7528.000000   
25%    2.023061e+07  1.132500e+04   2.007100e+07  6.109300e+04     135.000000   
50%    2.023063e+07  2.269500e+04   2.012070e+07  1.141370e+05     654.000000   
75%    2.023071e+07  5.906800e+04   2.016063e+07  1.748540e+05    3059.750000   
max    2.023080e+07  1.533690e+05   9.091041e+07  9.977387e+06   85021.000000   

                PLZ  Fzg_anlage_datum  Belegerstelltam  Beleggedrucktam  \
count  1.305237e+06      1.309302e+06     1.307576e+06     1.220883e+06   
mean   4.276390e+04      2.019603e+07     2.022572e+07     2.023020e+07   
std    2.409396e+04      5.556088e+04 

Bei der Tabelle PosArt konnte beobachetet werden:

1. Die Daten scheinen gleich mit Tabelle Belege zu sein

Unterschied zwischen Tabelle Belege und Tabelle PosArt prüfen:

In [19]:
diff = df_Belege[df_Belege != df_PosArt].stack()
print(diff)

Series([], dtype: object)


Die Tabellen beinhalten gleiche Daten.